src/mbgl/gl/attribute.cpp

```diff
@@ -61,131 +64,163 @@ template class VariableAttributeBinding<float, 4>;
 template <>
 void ConstantAttributeBinding<uint8_t, 1>::bind(Context&, AttributeLocation location, optional<VariableAttributeBinding<uint8_t, 1>>& oldBinding, std::size_t) const {
     assert(location != 0);
-    oldBinding = {};
-    MBGL_CHECK_ERROR(glDisableVertexAttribArray(location));
+    if (oldBinding) {
+        MBGL_CHECK_ERROR(glDisableVertexAttribArray(location));
+        oldBinding = {};
+    }
```

Comment:
This fails when we're not using VAOs. In that case, we're [resetting the binding](https://github.com/mapbox/mapbox-gl-native/blob/96b30a030f9c627fff1072b4a1e86c67d11af9f8/src/mbgl/gl/segment.hpp#L50) for every segment we're drawing to force a rebind, so `oldBinding` will always be empty and we may fail to disable the vertex attribute array for a certain location.